SDK/AppCenterCrashes/Microsoft.AppCenter.Crashes.Windows.Shared/Crashes.cs

```diff
@@ -213,9 +214,14 @@ private Task HandleUserConfirmationAsync()
                 tasks.Add(Channel.EnqueueAsync(log));
                 _unprocessedManagedErrorLogs.Remove(key);
                 ErrorLogHelper.RemoveStoredErrorLogFile(key);
-                var errorReport = new ErrorReport(log, null);
-                var attachments = GetErrorAttachments?.Invoke(errorReport);
-                tasks.Add(SendErrorAttachmentsAsync(log.Id, attachments));
+                if (GetErrorAttachments != null)
+                {
+                    var errorReport = new ErrorReport(log, null);
+
+                    // This must never called while a lock is held.
+                    var attachments = GetErrorAttachments?.Invoke(errorReport);
```

Comment:
Redundant.
```suggestion
                    var attachments = GetErrorAttachments.Invoke(errorReport);
```

Reply:
Not redundant because it can be unset by the time line 222 is reached

Reply:
Then why do you check it in L217?  You can get errorReport and get attachment, only add a task when attachment isn't null.

Reply:
Because there is no need to create an error report unless the callback is set. The race condition is not as harmful in the other direction, just trying to avoid a crash. I think I have a better way that will be more clear though.

Reply:
Improved.